Zend/zend_types.h

```diff
@@ -370,9 +384,28 @@ struct _zend_resource {
 	void             *ptr;
 };
 
+typedef struct _zend_property_info zend_property_info;
+
+typedef struct {
+	size_t num;
+	size_t num_allocated;
+	zend_property_info *ptr[1];
+} zend_property_info_list;
+
+typedef union {
+	zend_property_info *ptr;
+	uintptr_t list;
+} zend_property_info_source_list;
+
+#define ZEND_PROPERTY_INFO_SOURCE_FROM_LIST(list) (0x1 | (uintptr_t) (list))
+#define ZEND_PROPERTY_INFO_SOURCE_TO_LIST(list) ((zend_property_info_list *) ((list) & ~0x1))
+#define ZEND_PROPERTY_INFO_SOURCE_IS_LIST(list) ((list) & 0x1)
+
```

Comment:
In general, we shouldn't limit "sources" to properties info only.
Tomorrow, we'll get typed variables, arrays and shapes...
Can we don't track sources at all? (I understand, this will change the proposed semantic)